src/main/java/org/socialhistoryservices/delivery/request/controller/AbstractRequestController.java

```diff
@@ -268,13 +277,41 @@ protected int getMaxResults(Map<String, String[]> p) {
             try {
                 maxResults = Math.max(0, Math.min(Integer.parseInt(p.get("page_len")[0]),
                         deliveryProperties.getRequestMaxPageLen()));
-            } catch (NumberFormatException ex) {
+            }
+            catch (NumberFormatException ex) {
                 throw new InvalidRequestException("Invalid page length: " + p.get("page_len")[0]);
             }
         }
         return maxResults;
     }
 
+    protected List<Holding> getHoldingsWithSpecificStatus(ArrayList<String> pids, Holding.Status holdingStatus) {
```

Comment:
Don't do this on the controller level. But on the record or holding level, as it involves building a query.